app/models/ArrivalUpdate.scala

```diff
@@ -39,26 +72,35 @@ object MessageStatusUpdate extends MongoDateTimeFormats {
     )
 }
 
-case class ArrivalUpdate(arrivalUpdate: Option[ArrivalStatus], messageUpdate: Option[MessageStatusUpdate])
+final case class ArrivalStatusUpdate(arrivalStatus: ArrivalStatus) extends ArrivalUpdate
 
-object ArrivalUpdate {
+object ArrivalStatusUpdate extends MongoDateTimeFormats {
+  implicit def arrivalStatusUpdate(implicit writes: Writes[ArrivalStatus]): ArrivalModifier[ArrivalStatusUpdate] =
+    value =>
+      Json.obj(
+        "$set" -> Json.obj(
+          "status"      -> value.arrivalStatus,
+          "lastUpdated" -> LocalDateTime.now.withSecond(0).withNano(0)
+        ))
+}
 
-  implicit object ArrivalUpdateSemiGroupInst extends Semigroup[ArrivalUpdate] {
-    override def combine(x: ArrivalUpdate, y: ArrivalUpdate): ArrivalUpdate =
-      ArrivalUpdate(
-        y.arrivalUpdate orElse x.arrivalUpdate,
-        y.messageUpdate orElse x.messageUpdate
-      )
-  }
+final case class CompoundStatusUpdate(arrivalStatusUpdate: ArrivalStatusUpdate, messageStatusUpdate: MessageStatusUpdate) extends ArrivalUpdate
 
-  implicit object ArrivalUpdateArrivalModifier extends ArrivalModifier[ArrivalUpdate] {
-    override def toJson(a: ArrivalUpdate): JsObject = {
+object CompoundStatusUpdate {
+  implicit val arrivalUpdate: ArrivalModifier[CompoundStatusUpdate] =
+    csu => ArrivalModifier.toJson(csu.arrivalStatusUpdate) deepMerge ArrivalModifier.toJson(csu.messageStatusUpdate)
+}
 
-      val arrivalUpdateJson: JsObject = a.arrivalUpdate.map(ArrivalModifier.toJson[ArrivalStatus]).getOrElse(Json.obj())
-      val messageUpdateJson: JsObject = a.messageUpdate.map(ArrivalModifier.toJson[MessageStatusUpdate]).getOrElse(Json.obj())
+final case class ArrivalPutUpdate(movementReferenceNumber: MovementReferenceNumber, arrivalUpdate: CompoundStatusUpdate) extends ArrivalUpdate
 
-      arrivalUpdateJson deepMerge messageUpdateJson
-    }
-  }
+object ArrivalPutUpdate extends MongoDateTimeFormats {
+
+  implicit val arrivalPutUpdateArrivalModifier: ArrivalModifier[ArrivalPutUpdate] = a =>
+    Json.obj(
+      "$set" -> Json.obj(
+        "movementReferenceNumber" -> a.movementReferenceNumber,
+        "lastUpdated"             -> LocalDateTime.now.withSecond(0).withNano(0)
```

Comment:
Getting the LocalDateTime.now but manually setting the seconds and nano to 0. Are these required as we can set a formatter to remove them?

Reply:
I think that the setting the seconds and nano to 0 is a workaround for testing purposes, which is a bad practice. I'll raise a tech debt issue for this